src/quickfix.c

```diff
@@ -2325,6 +2342,8 @@ copy_loclist(qf_list_T *from_qfl, qf_list_T *to_qfl)
 	to_qfl->qf_index = 1;
     }
 
+    sync_loclist_ptr(to_qfl);
```

Comment:
Thanks for reporting the problem and the fix.

This problem happens because in the qf_jump_newwin() function,
only the qf_index field is updated before jumping to the file.
The qf_ptr field also should be updated before jumping to the file.
The following patch fixes this problem:

diff --git a/src/quickfix.c b/src/quickfix.c
index 4ac4a01a1..39424ca68 100644
--- a/src/quickfix.c
+++ b/src/quickfix.c
@@ -3461,6 +3461,7 @@ qf_jump_newwin(qf_info_T  *qi,
     }
 
     qfl->qf_index = qf_index;
+    qfl->qf_ptr = qf_ptr;
     if (qf_win_pos_update(qi, old_qf_index))
        // No need to print the error message if it's visible in the error
        // window

With this change, there is no need to iterate through all the
location list entries to find the pointer to the current entry.

Reply:
Thanks for the tip! I'll use this solution.